src/tests/index_btree_benchmark.cpp

```diff
@@ -0,0 +1,396 @@
+        run_in_parallel([this](uint32_t i) { do_put(i, btree_put_type::INSERT); });
+    }
+
+    void get_random_in_parallel() {
+        reset_op_num();
+        run_in_parallel([this](uint32_t i) { do_get(i); });
+    }
+
+    void update_random_in_parallel() {
+        reset_op_num();
+        run_in_parallel([this](uint32_t i) { do_put(i, btree_put_type::UPDATE); });
+    }
+
+    void remove_random_in_parallel() {
+        reset_op_num();
+        run_in_parallel([this](uint32_t i) { do_remove(i); });
+    }
+
+    uint32_t get_op_num() const { return m_num_ops.load(); }
+
+    uint32_t get_thread_num() const { return num_threads; }
+
+private:
+    void setup() {
+        test_common::HSTestHelper::start_homestore(
+            "index_btree_benchmark", {{HS_SERVICE::META, {.size_pct = 10.0}}, {HS_SERVICE::INDEX, {.size_pct = 70.0}}});
+
+        auto uuid = boost::uuids::random_generator()();
+        auto parent_uuid = boost::uuids::random_generator()();
+
+        auto cfg = BtreeConfig(hs()->index_service().node_size());
+        cfg.m_leaf_node_type = T::leaf_node_type;
+        cfg.m_int_node_type = T::interior_node_type;
+        if (SISL_OPTIONS.count("disable_merge")) { cfg.m_merge_turned_on = false; }
+        m_btree = std::make_shared< typename T::BtreeType >(uuid, parent_uuid, 0, cfg);
+        hs()->index_service().add_index_table(m_btree);
+    }
+
+    void teardown() { test_common::HSTestHelper::shutdown_homestore(); }
+
+    void do_put(uint64_t k, btree_put_type put_type) {
+        V value = V::generate_rand();
+        K key = K{k};
+        auto sreq = BtreeSinglePutRequest{&key, &value, put_type};
+        auto ret = m_btree->put(sreq);
+        HS_REL_ASSERT_EQ(ret, btree_status_t::success, "put failed");
+    }
+
+    void do_get(uint64_t k) {
+        K key = K{k};
+        V value;
+        auto req = BtreeSingleGetRequest{&key, &value};
+        auto ret = m_btree->get(req);
+        HS_REL_ASSERT_EQ(ret, btree_status_t::success, "get failed");
+    }
+
+    void do_remove(uint32_t k) {
+        V value;
+        K key = K{k};
+        auto rreq = BtreeSingleRemoveRequest{&key, &value};
+        auto ret = m_btree->remove(rreq);
+        HS_REL_ASSERT_EQ(ret, btree_status_t::success, "remove failed {}", ret);
+    }
+
+    void run_in_parallel(function< void(uint32_t) > cb) {
+        num_threads = SISL_OPTIONS["num_threads_for_parallel"].as< uint32_t >();
+        const auto num_entries = SISL_OPTIONS["num_entries"].as< uint32_t >();
+        std::vector< uint32_t > vec(num_entries * num_threads);
+        iota(vec.begin(), vec.end(), 0);
+        std::random_shuffle(vec.begin(), vec.end());
+        std::condition_variable done_cv;
+        std::mutex done_mtx;
+        auto thread_count{num_threads};
+
+        for (uint i{0}; i < num_threads; ++i) {
+            auto start = i * num_entries;
+            auto end = (i + 1) * num_entries;
+            std::thread([&, this, start, end]() {
```

Comment:
We also support fibers. Please use iomgr instead of std::thread

Reply:
ack